File: 闭包oc/闭包oc/ViewController.m
//
//  ViewController.m
//  闭包oc
//
//  Created by 李奕权 on 2019/10/22.
//  Copyright © 2019 李奕权. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}

-(void) loadData(){
    dispatch_sync(dispatch_get_global_queue(0, 0), {()-> void in
        
    })
}


@end

File: 闭包的演练/闭包的演练/ViewController.swift
//
//  ViewController.swift
//  闭包的演练
//
//  Created by 李奕权 on 2019/10/22.
//  Copyright © 2019 李奕权. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        
        
        
        
//        loadData()
        
    }
    
    
    //加载数据
    func loadData(){
        let queue = DispatchQueue(label: "hwhQuene1",attributes: .concurrent)
        
        queue.sync {
            for _ in 0...2{
                Thread.sleep(forTimeInterval: 2)
                print("线程1运行---%@",Thread.current)
            }
        }
        
        queue.async {
            for _ in 0...2{
                Thread.sleep(forTimeInterval: 2)
                print("线程2运行---%@",Thread.current)
            }
            
        }
        
        queue.sync {
            for _ in 0...2{
                Thread.sleep(forTimeInterval: 2)
                print("线程3运行---%@",Thread.current)
            }
            
        }
    }


}

